### PR TITLE
Duplicate prescription button disabled if lacking permissions

### PR DESCRIPTION
**Pull request by captDaylight**
https://www.notion.so/photons/Duplicate-prescription-button-should-be-disabled-if-i-can-t-prescribe-ba7ffb415189428c9f98602fa43028fd?pvs=4

### CODE REVIEW

File: apps/app/src/hooks/usePermissions.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { usePhoton } from '@photonhealth/react';
+import jwtDecode from 'jwt-decode';
+import { Permission } from 'packages/sdk/dist/types';
+import { useEffect, useState } from 'react';
+
+function checkHasPermission(subset: Permission[], superset: Permission[]) {
```
Comment (jknapp25) [inline apps/app/src/hooks/usePermissions.ts:6]: is this the same idea?

`return subset.every(permission => superset.includes(permission));`

Comment (captDaylight) [reply apps/app/src/hooks/usePermissions.ts:6]: love it

File: apps/app/src/views/routes/Prescription.tsx [hunk 2]
```diff
@@ -68,6 +69,7 @@ export const CANCEL_PRESCRIPTION = gql`
 export const Prescription = () => {
   const params = useParams();
   const id = params.prescriptionId;
+  const hasPermission = usePermissions(['write:prescription']);
```
Comment (jknapp25) [inline apps/app/src/views/routes/Prescription.tsx:72]: Does write:prescription also require auth check?

**Review by jknapp25 — APPROVED** [review]
Cool!

File: apps/app/src/views/routes/Prescription.tsx (hunk 2 above)
Comment (captDaylight) [reply apps/app/src/views/routes/Prescription.tsx:72]: not sure I follow, these permissions are coming available after authentication

**Pull request merged**